asvdb/asvdb.py

```diff
@@ -925,49 +943,117 @@ def __createLockfile(self, dirPath):
     ###########################################################################
     # S3 Locking methods
     ###########################################################################
-    def __getS3Lock(self, url):
-        # FIXME: write this
-        raise NotImplementedError
+    def __getS3Lock(self):
+        thisLockfile = path.join(self.bucketKey, self.lockfileName)
+        # FIXME: This shouldn't be needed? But if so, be smarter about
+        # preventing an infintite loop?
+        i = 0
+        otherLockfileTimes = []
+        while i < 1000:
+            otherLockfileTimes = self.__updateS3LockfileTimes()
+            debugCounter = 0
 
+            while otherLockfileTimes:
+                debugCounter += 1
+                if self.debugPrint:
+                    print(f"This lock file will be {thisLockfile} but other "
+                          f"locks present: {otherLocks}, waiting to try to "
+                          "lock again...")
+                time.sleep(2)
+                otherLockfileTimes = self.__updateOtherLockfileTimes()
```

Comment:
is the `__updateOtherLockfileTimes` method safe to use with S3? I noticed it uses the `glob` library which looks like it's specific to local filesystems.